### PR TITLE
Add compatibility with macOS

### PR DESCRIPTION
**Pull request by SMillerDev**
This patch has been needed for ages on macOS and I finally got around to upstreaming it.

### CODE REVIEW

File: config.m4 [hunk 1]
```diff
@@ -19,15 +21,29 @@ if test "$PHP_UUID" != "no"; then
...
+#error compiling on darwin
+#endif
+],
+[],
+[export UUID_ON_MAC=true])
+
+
+  if test $UUID_ON_MAC != "true"; then
+    PHP_SUBST(UUID_SHARED_LIBADD)
+    PHP_ADD_LIBRARY_WITH_PATH(uuid, $PHP_UUID_DIR/lib, UUID_SHARED_LIBADD)
```
Comment (remicollet) [inline config.m4]: Please use `$PHP_LIBDIR`

Comment (SMillerDev) [reply config.m4]: Like this?
```suggestion
    PHP_ADD_LIBRARY_WITH_PATH(uuid, $PHP_LIBDIR, UUID_SHARED_LIBADD)
```

Comment (remicollet) [reply config.m4]: See deleted line above... `$PHP_UUID_DIR/$PHP_LIBDIR`

File: uuid.c [hunk 2]
```diff
@@ -377,8 +378,10 @@ PHP_FUNCTION(uuid_time)
 	RETURN_LONG(uuid_time(u, NULL));
 }
 /* }}} uuid_time */
+#endif /* HAVE_UUID_TIME */
 
 
+#if HAVE_UUID_MAC
```
Comment (remicollet) [inline uuid.c:384]: No, HAVE_UUID_MAC is never defined, this is not a library symbol

**Comment by remicollet** [comment]
Linux build is totally broken 

```
/home/php/git/uuid/uuid_arginfo.h:102:27: erreur: « arginfo_uuid_mac » non déclaré ici (hors de toute fonction); vouliez-vous utiliser « arginfo_uuid_time » ?
  102 |         ZEND_FE(uuid_mac, arginfo_uuid_mac)
      |                           ^~~~~~~~~~~~~~~~
/usr/include/php/Zend/zend_API.h:77:80: note: dans la définition de la macro « ZEND_RAW_FENTRY »
   77 | #define ZEND_RAW_FENTRY(zend_name, name, arg_info, flags)   { zend_name, name, arg_info, (uint32_t) (sizeof(arg_info)/sizeof(struct _zend_internal_arg_info)-1), flags },
      |                                                                                ^~~~~~~~
/home/php/git/uuid/uuid_arginfo.h:102:9: note: dans l'expansion de la macro « ZEND_FE »
  102 |         ZEND_FE(uuid_mac, arginfo_uuid_mac)
      |         ^~~~~~~
/home/php/git/uuid/uuid_arginfo.h:103:29: erreur: « arginfo_uuid_parse » non déclaré ici (hors de toute fonction); vouliez-vous utiliser « arginfo_uuid_time » ?
  103 |         ZEND_FE(uuid_parse, arginfo_uuid_parse)
      |                             ^~~~~~~~~~~~~~~~~~
/usr/include/php/Zend/zend_API.h:77:80: note: dans la définition de la macro « ZEND_RAW_FENTRY »
   77 | #define ZEND_RAW_FENTRY(zend_name, name, arg_info, flags)   { zend_name, name, arg_info, (uint32_t) (sizeof(arg_info)/sizeof(struct _zend_internal_arg_info)-1), flags },
      |                                                                                ^~~~~~~~
/home/php/git/uuid/uuid_arginfo.h:103:9: note: dans l'expansion de la macro « ZEND_FE »
  103 |         ZEND_FE(uuid_parse, arginfo_uuid_parse)
      |         ^~~~~~~
/home/php/git/uuid/uuid_arginfo.h:104:31: erreur: « arginfo_uuid_unparse » non déclaré ici (hors de toute fonction); vouliez-vous utiliser « arginfo_uuid_compare » ?
  104 |         ZEND_FE(uuid_unparse, arginfo_uuid_unparse)
      |                               ^~~~~~~~~~~~~~~~~~~~
/usr/include/php/Zend/zend_API.h:77:80: note: dans la définition de la macro « ZEND_RAW_FENTRY »
   77 | #define ZEND_RAW_FENTRY(zend_name, name, arg_info, flags)   { zend_name, name, arg_info, (uint32_t) (sizeof(arg_info)/sizeof(struct _zend_internal_arg_info)-1), flags },
      |                                                                                ^~~~~~~~
/home/php/git/uuid/uuid_arginfo.h:104:9: note: dans l'expansion de la macro « ZEND_FE »
  104 |         ZEND_FE(uuid_unparse, arginfo_uuid_unparse)
      |         ^~~~~~~
make: *** [Makefile:208 : uuid.lo] Erreur 1

```

File: config.m4 [hunk 3]
```diff
@@ -6,6 +6,8 @@ if test "$PHP_UUID" != "no"; then
   for dir in $PHP_UUID /usr/local /usr; do
     if test -f "$dir/include/uuid/uuid.h"; then
       PHP_UUID_DIR="$dir"
+    elif test -f "$dir/include/ossp/uuid.h"; then
+      PHP_UUID_DIR="$dir"
```
Comment (remicollet) [inline config.m4:10]: probably simpler to define UUID_ON_MAC here (rather than using AC_TRY_COMPILE)

File: config.m4 [hunk 4]
```diff
@@ -19,15 +21,29 @@ if test "$PHP_UUID" != "no"; then
   PHP_CHECK_FUNC_LIB(uuid_generate_md5, uuid)
   PHP_CHECK_FUNC_LIB(uuid_generate_sha1, uuid)
 
+  PHP_CHECK_FUNC_LIB(uuid_time, uuid)
+  PHP_CHECK_FUNC_LIB(uuid_mac, uuid)
```
Comment (remicollet) [inline config.m4:25]: doesn't make sense, this is not a library symbol

File: uuid_arginfo.h [hunk 5]
```diff
@@ -42,17 +42,21 @@ ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(arginfo_uuid_variant, 0, 1, IS_LONG, 0)
...
 
+#if defined(HAVE_UUID_MAC)
 ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(arginfo_uuid_mac, 0, 1, IS_STRING, 0)
 	ZEND_ARG_TYPE_INFO(0, uuid, IS_STRING, 0)
 ZEND_END_ARG_INFO()
 
 #define arginfo_uuid_parse arginfo_uuid_mac
 
 #define arginfo_uuid_unparse arginfo_uuid_mac
+#endif
```
Comment (remicollet) [inline uuid_arginfo.h:59]: Bad position.

Please don't change the arginfo.h files, but generate them  (automatically when stub.php is altered, with PHP >= 8)

File: config.m4 (hunk 3 above)
Comment (SMillerDev) [reply config.m4:10]: There is no guarantee that this is on mac though.

Comment (SMillerDev) [reply config.m4:10]: AFAIK anyone can install http://www.ossp.org/pkg/lib/uuid/

File: uuid_arginfo.h (hunk 5 above)
Comment (SMillerDev) [reply uuid_arginfo.h:59]: It's not clear to me how to do that. Do I run the `stub.php`? `phpize`?

Comment (remicollet) [reply uuid_arginfo.h:59]: When you change the stub, a simple make will regenerate the arginfo.

BTW this case can be a bit ugly... perhaps need to move conditional methods at the end to ensure all arginfo are there.

File: config.m4 (hunk 3 above)
Comment (remicollet) [reply config.m4:10]: But, if this is a lib, you need to add them in builder options.....

**Comment by remicollet** [comment]
Closing as obviously not ready.
Please open a new clean PR when ready.

**Pull request closed without merging**